### DISCUSSION TITLE
Differing Temperatures From API vs Web

### DISCUSSION THREAD

**Discussion by jkreski** [weather-gov/api#729] [General]
I am doing a fetch of most current temperature observation from:  https://api.weather.gov/stations/KMBL/observations/latest and I am checking the accuracy by comparing it to what I see at: https://forecast.weather.gov/MapClick.php?CityName=Beulah&state=MI&site=APX&textField1=44.6296&textField2=-86.0955&e=0

I am seeing significant differences at times and am wondering why this is or if I am missing something in my approach.  Does anyone else notice this?  Thanks!

**Comment by brewin** [comment]
In my experience, the /observations/latest endpoint is usually delayed by an hour or more, which makes it wildly inaccurate much of the time. There are other issues with it too. https://github.com/weather-gov/api/discussions/726

**Reply by scadergit** [reply]
Unlike mapclick that has limited stations, the API uses MADIS to provide observation data. MADIS can take up to 20m from the observation time to run quality checks. For a station that makes an observation on the hour, that can be 80m from the previous observation. That tends to be the extreme unless another issue is present.

**Reply by brewin** [reply]
I think that aligns with what I'm seeing. Typically the /observations/latest timestamp is 30 to 120 minutes old. For example, at 10:45 AM CDT, /KMCI/observations/latest timestamp was 8:53, almost 2 hours old. At 10:47, it was updated to 9:53. So, the new "latest" was already almost an hour old. My expectation is that /latest would always be updated within a few minutes of the observation time, like the MapClick API has done over the years. But I know MapClick has had its own issues.

**Reply by jkreski** [reply]
What time zone are the time stamps in?  UTC, EST, CST, etc?

**Reply by scadergit** [reply]
I believe they are all UTC, as indicated by `+00:00`

**Reply by brewin** [reply]
They are UTC. I converted them to the local time in my comment. If you use the forecast endpoint, it has timestamps with UTC offsets.

**Comment by jkreski** [comment]
Is there a better link to use?  Or use a  different service's API?

**Reply by brewin** [reply]
There is the [aviationweather.gov](https://aviationweather.gov/data/api/) metar endpoint. You have to calculate a bounding box and parse the metar. I have not used it, but I'm considering it.

**Comment by scadergit** [comment]
Will note here as well that the beta weather.gov 2.0 site uses the API, so a lot of these issues are getting attention internally. Won't be fixed tomorrow, but it's recognized this isn't to standard.
